syncer/syncer.go

```diff
@@ -0,0 +1,235 @@
+package syncer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/SkynetLabs/blocker/blocker"
+	"github.com/SkynetLabs/blocker/database"
+	"github.com/SkynetLabs/blocker/skyd"
+	"github.com/sirupsen/logrus"
+	"gitlab.com/NebulousLabs/errors"
+	"go.sia.tech/siad/build"
+	"go.sia.tech/siad/crypto"
+)
+
+var (
+	// syncInterval defines the amount of time between syncs of external
+	// portal's blocklists, which can be defined in the environment using the
+	// key BLOCKER_SYNC_LIST
+	syncInterval = build.Select(
+		build.Var{
+			Dev:      time.Minute,
+			Testing:  time.Second,
+			Standard: 4 * time.Hour,
```

Comment:
This seems like a bit too much considering that we have had providers request that we block certain content within an hour. I do understand that we don't like fetching that much data at once though.

So one thought might be to extend skyd itself to also provide a timestamp of the last time we updated the blocklist. Then you poll that more frequently and only if it changed we fetch the whole thing again.

Reply:
I think I should've done a better job explaining the idea in the PR. The idea for the blocklist sync is not to keep up to date with the local skyd instance, but rather to bootstrap a portal with the blocklist of another portal. So really nothing changes and the portal operator should still run the abuse software to ensure blocking happens immediately, which it will.

This 4 hour sync time is merely an interval with which we fetch the blocklist from another portal. The idea of adding a timestamp to the blocklist endpoint though is great, we should definitely do that. I just don't see the issue with the 4h interval here. Maybe you don't see one either now that it's more clear what the feature is/does?

Reply:
I thought about this more and I have come to see your point. I think this is a F/U issue as it needs an update to skyd but it's definitely one that we want to make. After that we'd want to indeed lower this to something like every 15 minutes or so I'd say.